OCD-4347 - Remove "unattested criteria" from Listings in data

tmy1313 opened the pull request.

Commit:
feat: stop creating certification results for unattested to criteria

OCD-4349

Commit:
fix: temporarily disable broken tests

OCD-4349

Commit:
test: update tests to match new reuirements

OCD-4349

Commit:
feat: display a FALSE rather than ? when the unattested criteria does not exist

OCD-4349

Commit:
Merge branch 'staging' into OCD-4347

Commit:
feat: explicitly remove criteria where succes is null or false

OCD-4347

andlar commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/normalizer/CertificationCriterionNormalizer.java:
```diff
@@ -27,52 +23,18 @@
 
     public void normalize(CertifiedProductSearchDetails listing) {
-        addEditionCriteriaNotPresentInListing(listing);
+        listing.getCertificationResults().removeIf(cr -> cr.isSuccess() == null || !cr.isSuccess());
```
Does this normalizer run on 2014 Edition Listings when/if they're edited? Because if it does, there are some 2014 Criteria that have g1Success or g2Success values as "true" that still need to show on the Listing, even if the Listing doesn't attest to the criteria

tmy1313 replied:
It only runs on uploaded listings.

kekey1 replied:
I'm pretty sure the normalizers all run on both uploaded and editing listings. I see the ListingDetailsNormalizer.normalize method being called on upload and update.

Commit:
test: remove invalid test

OCD-4347

Commit:
feat: not remove unattested criteria for 2011 and 2014 listings when saving

OCD-4347

Commit:
fix: only check 2014 listings w/ g1 or g2 success

OCD-4347

Commit:
fix: fix copy paste error, 2011 s/b 2014

OCD-4347

Commit:
fix: add null check

OCD-4347

kekey1 committed:
fix: Change logic for criteria removed questionable activity

[#OCD-4347]

Previously we could assume that an 'origListing' and a 'newListing' had
the same set of certification results with some attested and some
unattested (using the 'success' boolean). With this ticket, we can no
longer make that assumption and have to calculate whether criteria were
added or removed differently.

[#OCD-4347]

kekey1 committed:
fix: Adjust criteria added/removed calculations for subscriptions

[#OCD-4347]

Previously we could assume that an 'origListing' and a 'newListing' had
the same set of certification results with some attested and some
unattested (using the 'success' boolean). With this ticket, we can no
longer make that assumption and have to calculate whether criteria were
added or removed differently.

[#OCD-4347]

kekey1 committed:
fix: Modify editing logic for adding/removing cert results

[#OCD-4347]

Previously we could assume that an 'origListing' and a 'newListing' had
the same set of certification results with some attested and some
unattested (using the 'success' boolean). With this ticket, we can no
longer make that assumption and have to calculate whether criteria were
added or removed differently.

[#OCD-4347]

kekey1 committed:
Merge branch 'staging' into OCD-4347

tmy1313 committed:
Merge pull request #16 from kekey1/OCD-4347

OCD-4347: Unattested criteria removal updates for editing, reporting

Commit:
Merge branch 'staging' into OCD-4347

Commit:
Merge branch 'OCD-4347' of https://github.com/tmy1313/chpl-api into OCD-4347

Commit:
Merge branch 'staging' into OCD-4347

Commit:
fix: properly save unattested to criteria for 2014 listings

OCD-4347

Commit:
fix: handle when the unattested criteria does not exist in the Cure Report job/curesStatistics/CertificationResultActivityHistoryHelper

OCD-4347
